client/layout/guided-tours/index.js

```diff
@@ -23,22 +24,48 @@ class GuidedTours extends Component {
 		return this.props.tourState !== nextProps.tourState;
 	}
 
-	next = ( tour, nextStepName ) => {
+	next = ( { tour, tourVersion, nextStepName, skipping = false } ) => {
+		// Don't immediately record the transition to the next step. Instead,
+		// remember the name of the step and _maybe_ record it later (at the
+		// next invokation of `next` or `quit`), depending on whether the step
+		// was skipped (see Step#skipIfInvalidContext).
```

Comment:
The recording happens immediately, but what I meant by the comment was that the recording happens for the _previous_/_current_ (depending on the perspective) step, and not `nextStepName`.